### DISCUSSION TITLE
SpeakerSegmentation pipeline: stitching?

### DISCUSSION THREAD

**Discussion by stevenhillis** [pyannote/pyannote-audio#870] [Q&A] [answered]
Hello! I'm experimenting with the V2 speaker segmentation pipeline. I don't quite understand what's happening after the 3D SlidingWindowFeature <code>segmentations</code> output from the model is produced. What is happening with the stitching? And why might I want to skip_stitching and just binarize the outputs?

**Comment by hbredin** [comment, marked as answer]
The idea is to try and solve the speaker permutation problem (the same speaker in two consecutive chunks may activate two different classes). This is still very much a work in progress though.

**Reply by stevenhillis** [reply]
That makes sense, thank you @hbredin! Have you tried overlapping the chunks and using the two sets of labels for the  overlapped frames to solve this permutation problem?

**Reply by hbredin** [reply]
That is more or less what is currently implemented.
